Updating sequencer config implementation (#330)
astriaorg/astria: crates/astria-sequencer/README.md

# Astria-Sequencer

## Dependencies

We use [just](https://just.systems/man/en/chapter_4.html) for convenient project
specific commands.

- rust 1.68
- gcc-12, gcc-12-libs
- go 1.18+

### Penumbra

Because `penumbra-storage:0.54.1` depends on `rocksdb:0.19.0` compilation on gcc
13 will not work.

On arch linux:

> ```bash

Reviewer: ```suggestion
```sh
```

same with others in this file for consistency

<https://github.com/rust-rocksdb/rust-rocksdb/issues/713>
<https://github.com/facebook/rocksdb/pull/11118>

## Usage

## Running the Sequencer

### Configuration

Composer is configured via environment variables. An example configuration can
be seen in `local.env.example`.

To copy a configuration to your `.env` file run:

```sh

# Can specify an environment
just copy-env <ENVIRONMENT>

# By default will copy `local.env.example`
just copy-env
```

### Install cometbft

Ensure `~/go` is in your `PATH`, or `GOPATH` is set to some other place in your
`PATH`.

```sh

> git clone https://github.com/astriaorg/cometbft

Reviewer 2: Don't rm this lin

<https://github.com/rust-rocksdb/rust-rocksdb/issues/713>
<https://github.com/facebook/rocksdb/pull/11118>

## Running the Sequencer

### Configuration

Composer is configured via environment variables. An example configuration can
be seen in `local.env.example`.

To copy a configuration to your `.env` file run:

```sh

# Can specify an environment
just copy-env <ENVIRONMENT>

# By default will copy `local.env.example`
just copy-env
```

### Install cometbft

Ensure `~/go` is in your `PATH`, or `GOPATH` is set to some other place in your
`PATH`.

```sh

> just install-cometbft

Reviewer 2: There is no such just command. Also, I would probably leave the steps in place. We don't want to mess with people's setup (e.g. if they have a different GOPATH, like myself).

> ../../target/debug/astria-sequencer --db-filepath=/tmp/astria_db

Author: Good catch

```sh
# initialize the node
cometbft init

> # set the block time to 14s

Reviewer: ```suggestion
# set the block time to 15s
```

> sed -i'.bak' 's/timeout_commit = "0s"/timeout_commit = "15s"/g' ~/.cometbft/config/config.toml

Reviewer: why is this changed?

# start the node
cometbft node
```

You should see blocks being produced.

You can also use `just` to run the above commands:

```bash
just run-cometbft
```

> ## Testnet

Reviewer: I kinda liked having the testnet instructions in a different file, maybe just add a link to it here instead?

```sh
# initialize the node
cometbft init

# inside astria-sequencer, update the genesis file to include genesis application state
../../target/debug/astria-sequencer-utils --genesis-app-state-file=test-genesis-app-state.json  --destination-genesis-file=$HOME/.cometbft/config/genesis.json

# set the block time to 15s
sed -i'.bak' 's/timeout_commit = "1s"/timeout_commit = "15s"/g' ~/.cometbft/config/config.toml

# start the node
cometbft start
```

You should see blocks being produced.

You can also use `just` to run the above commands:

```sh
just run-cometbft
```

## Testnet

> Check out the `TESTNET.md` file for details on how to run the astria testnet

Reviewer: ```suggestion
Check out the `TESTNET.md` file for details on how to run a multi-node sequencer testnet.
```
